### PR TITLE
Terrain detail and macro materials tutorial

### PR DESCRIPTION
**Pull request by cgalvan**
## Change summary

New tutorial to explain techniques for using macro materials and detail mapping together.

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

Signed-off-by: Chris Galvan <chgalvan@amazon.com>

### CODE REVIEW

**Comment by chanmosq** [comment]
Requesting a technical review from @sig-content-reviewers as well!

**Review by micronAMZN — CHANGES_REQUESTED** [review]
Nice work. I added a bunch of suggestions for style guidance and table formatting.

**Comment by micronAMZN** [comment]
On hold for image updates.

File: content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+---
+linkTitle: Understanding Frequency Separation
+title: Understanding Frequency Separation
+description: Learn various frequency separation techniques and how they apply to detail materials.
+weight: 300
+toc: true
+---
+
+This tutorial explains how to use frequency separation to create textures for terrain in **Open 3D Engine (O3DE)**. This is an image technique which is often used in retouching photos. However, it can also be a useful technique to understand and apply technically to the creation of textures and/or shading techniques.
```
[the remark this answers is not in the document]
Comment (cgalvan) [reply content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md:9]: I'll re-work this intro, thanks!

File: content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+## Generating low and high pass
+
+Terrain detail maps can be derived from running a high pass filter on an image. Starting from an image with lots of color and detail, we are going to separate out the low frequency and high frequency data.
+Separating low/high frequency data from an image is a manual process in an image editor, such as Photoshop.
+
+Here is the original image:
```
[the remark this answers is not in the document]
Comment (cgalvan) [reply content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md:16]: Sounds good!

File: content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+![Photoshop: Set blend mode of high pass to Linear light](/images/learning-guide/tutorials/environments/detail_macro_materials/ps_linear_light.png)
+
+As you can see in the screenshot, we've restored the original image fidelity by properly blending the low + high pass frequencies back together.
+
+## Downsampled low pass
+
+With the blending of low pass + high pass, we can actually downsample the low pass image and still achieve a final blended image with little perceptible loss in quality.
+We will downsample the low pass to a much smaller image, then reconstruct the image from two maps of different resolutions. Theoretically, we will loose some information which may decrease aspects like the fidelity, quality and overall data integrity of the image, but we can experiment with the levels to see where the quality of the final image starts to breakdown.
+
+| Original Low Pass (1024 x 1024 pixels) | Downsampled (64 x 64 pixels) | Interpolated (Bilinear) | Interpolated + Original High Pass (Reconstructed blend) | Difference (Original minus Reconstructed blend) |
```
[the remark this answers is not in the document]
Comment (cgalvan) [reply content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md:56]: Thanks!

**Comment by cgalvan** [comment]
I'm in the process of incorporating Mike's feedback, plus new images and some updated sections from Jonny.

File: content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+### Tiling images
+
+If you are working with tiling images, you will want to use the following steps in order to make sure that both the low pass and high pass will continue to properly tile after the frequency splitting happens:
```
[the remark this answers is not in the document]
Comment (cgalvan) [reply content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md:24]: I like this change, added!

File: content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  * Save the base image as a pattern
+* Image > Canvas Size
+  * Width/Height 300%
+* Edit > Fill ...
+  * Contents: Patterns
+  * Custom Pattern: (select the pattern you made previously from the base image)
+  * Note: This fills the image with the pattern. Consider it a 3x3 tiled version of your image, so we can work on the center tile.
+* Now repeat the steps above for splitting the image frequencies
+* Image > Canvas Size
+  * Width/Height: 33.33%, or the <original image size> in pixels
```
[the remark this answers is not in the document]
Comment (cgalvan) [reply content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md:37]: Thanks this looks great!

File: content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+These steps allow the gaussian blur to take into account the wrapped tiling along the adjacent borders.
+Now, when the low pass (blur) is calculated, the pixel information wraps. Then, when it's subtracted and cropped both the low pass and the high pass will still tile properly.
```
[the remark this answers is not in the document]
Comment (cgalvan) [reply content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md:40]: Thanks this is more informative!

File: content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+## Blending
+
+Next, we will combine (blend) the low pass (Layer 2) and the high pass (Layer 3).
+* In Photoshop, set the layer blend mode of the high pass to Linear light
+
+![Photoshop: Set blend mode of high pass to Linear light](/images/learning-guide/tutorials/environments/detail_macro_materials/ps_linear_light.png)
+
+As you can see in the screenshot, we've restored the original image fidelity by properly blending the low + high pass frequencies back together.
```
[the remark this answers is not in the document]
Comment (cgalvan) [reply content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md:49]: Good idea, incorporated!

File: content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+## Downsampled low pass
+
+With the blending of low pass + high pass, we can actually downsample the low pass image and still achieve a final blended image with little perceptible loss in quality.
+We will downsample the low pass to a much smaller image, then reconstruct the image from two maps of different resolutions. Theoretically, we will loose some information which may decrease aspects like the fidelity, quality and overall data integrity of the image, but we can experiment with the levels to see where the quality of the final image starts to breakdown.
```
[the remark this answers is not in the document]
Comment (cgalvan) [reply content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md:54]: Yeah this sounds better, updated!

File: content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+|-|-|-|-|-|
+|{{< image-width src="/images/learning-guide/tutorials/environments/detail_macro_materials/original_low_pass.png" width="150" alt="Original Low Pass (1024 x 1024 pixels)" >}}|{{< image-width src="/images/learning-guide/tutorials/environments/detail_macro_materials/downsampled_16.png" width="150" alt="Downsampled (16 x 16 pixels)" >}}|{{< image-width src="/images/learning-guide/tutorials/environments/detail_macro_materials/interpolated_16.png" width="150" alt="Interpolated (Bilinear)" >}}|{{< image-width src="/images/learning-guide/tutorials/environments/detail_macro_materials/blended_16.png" width="150" alt="Interpolated + Original High Pass (Reconstructed blend)" >}}|{{< image-width src="/images/learning-guide/tutorials/environments/detail_macro_materials/difference_16.png" width="150" alt="Difference (Original minus Reconstructed blend)" >}}|
+
+At this level of downsampling, we are starting to see a perceptible difference in integrity, and the quality is arguably diminished. However, the fidelity is still good enough to be useful for our terrain detail mapping use case.
```
[the remark this answers is not in the document]
Comment (cgalvan) [reply content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md:68]: Agreed, and yes good catch I think it should be "detail" so I'll change that as well.

File: content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+## Macro material low pass
+
+We can use the small downsampled low pass image in texturing for our Macro material. There are a several options for this use case:
```
[the remark this answers is not in the document]
Comment (cgalvan) [reply content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md:72]: Done!

File: content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+* Use the downsampled low pass image as a texture input for texturing
+* Use the downsampled low pass to generate a color ramp (and a matching height map), which can be used as input in programs like World Machine to use in colorization
+* Use the downsampled low pass as a color swatch for painting terrain
+
+Then with the proper blending and syncing the repeat, we can augment this with the the high pass detail texture and result in something similar to our original image up close.
```
[the remark this answers is not in the document]
Comment (cgalvan) [reply content/docs/learning-guide/tutorials/environments/understanding-frequency-separation.md:77]: Done!

**Review by HogJonny-AMZN — APPROVED** [review]

**Review by mbalfour-amzn — APPROVED** [review]

**Review by micronAMZN — APPROVED** [review]

**Pull request merged**